pkg/kube/controllers/extendedstatefulset/pod_mutator.go

```diff
@@ -160,7 +160,7 @@ func (m *PodMutator) addVolumeSpec(pod *corev1.Pod, volumeClaimTemplatesMap map[
 			_, foundVolumeClaimTemplate := volumeClaimTemplatesMap[volumeMount.Name]
 			if foundVolumeClaimTemplate {
 				podOrdinal := names.OrdinalFromPodName(pod.GetName())
-				persistentVolumeClaim := names.Sanitize(fmt.Sprintf("%s-%s-%s-%d", volumeMount.Name, "volume-management", extendedStatefulSetName, podOrdinal))
+				persistentVolumeClaim := fmt.Sprintf("%s-%s-%s-%d", volumeMount.Name, "volume-management", extendedStatefulSetName, podOrdinal)
```

Comment:
This doesn't need [Sanitize](https://github.com/cloudfoundry-incubator/quarks-utils/blob/master/pkg/names/names.go#L108)? It looks like it could go over the allowed character limit?

Reply:
> Kubernetes resources can have names up to 253 characters long.
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

I think the lower limit is only valid for pods.

Still it is possible that we hit that higher limit. But at this part of the code it's already to late to sanitize. We should check what happens if we create a `StatefulSet` with a `volumeClaimTemplate.name` length 253  as the StatefulSet controller will automatically append `-1`, `-2`, etc. which we also do here. So we can not use our own `Sanitize` logic, we have to do the exact same thing.

Reply:
Ah we had a similar issue with job names, where a digest was appended to the name when using the name for the job's pod.

https://github.com/cloudfoundry-incubator/quarks-utils/blob/master/pkg/names/names.go#L137-L139

Not sure if anything in kubecf will ever reach that higher limit, but to be safe we should allow for `-123`, if someone has a lot of instances maybe?

Reply:
I just tried to create a `StatefulSet` with `volumeClaimTemplate` with a very long (253 chars) name. No pod is ever created and `describe StatefulSet` shows error events. So we don't need to consider this case in the `PodMutator`, but handle it before creating the `volume-management`-`StatefulSet`.

I can fix this, but would prefer to do it in a separate PR. ok?

```yaml
---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: web
spec:
  selector:
    matchLabels:
      app: nginx
  serviceName: "nginx"
  template:
    metadata:
      labels:
        app: nginx
    spec:
      terminationGracePeriodSeconds: 10
      containers:
      - name: nginx
        image: k8s.gcr.io/nginx-slim:0.8
        ports:
        - containerPort: 80
          name: web
        volumeMounts:
        - name: wwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww
          mountPath: /usr/share/nginx/html
  volumeClaimTemplates:
  - metadata:
      name: wwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww
    spec:
      accessModes: [ "ReadWriteOnce" ]
      storageClassName: "default"
      resources:
        requests:
          storage: 1Gi
```

Reply:
yes, sounds good

Reply:
Lets first see what the outcome of [this](https://www.pivotaltracker.com/n/projects/2192232/stories/169531360) story is. If we can get rid of the versioned stateful sets we also won't need the volumeManagement anymore.